src/net.cpp

```diff
@@ -2539,15 +2539,44 @@ std::vector<CAddress> CConnman::GetAddresses()
     return addresses;
 }
 
-std::vector<CAddress> CConnman::GetAddresses(Network requestor_network)
+const std::vector<CAddress>& CConnman::GetAddresses(Network requestor_network, CAddress local_socket_address)
 {
+    auto local_socket_bytes = local_socket_address.GetAddrBytes();
+    uint64_t cache_id = GetDeterministicRandomizer(RANDOMIZER_ID_LOCALHOSTNONCE)
```

Comment:
Ever overlapping caches sounds dangerous. Why not just index a unique cache per bind address?